src/components/inputs/rich-text-input/rich-text-input.js

```diff
@@ -78,41 +81,65 @@ const plugins = [
   ListPlugin(),
 ];
 
-const RichTextInput = props => {
-  const { onChange } = props;
-  const onValueChange = ({ value }) => {
+class RichTextInput extends React.Component {
+  onValueChange = ({ value }) => {
     const event = {
-      target: { id: props.id, name: props.name, value },
+      target: {
+        id: this.props.id,
+        name: this.props.name,
+        value,
+      },
     };
-    onChange(event);
+    this.props.onChange(event);
   };
 
-  return (
-    <Editor
-      {...filterDataAttributes(props)}
-      id={props.id}
-      name={props.name}
-      disabled={props.isDisabled}
-      readOnly={props.isReadOnly}
-      value={props.value}
-      onFocus={props.onFocus}
-      onBlur={props.onBlur}
-      // we can only pass props to the Editor that Slate understands without getting
-      // warning in the console,
-      // so instead we pass our extra props through this `options` prop.
-      options={{
-        horizontalConstraint: props.horizontalConstraint,
-        defaultExpandMultilineText: props.defaultExpandMultilineText,
-        hasWarning: props.hasWarning,
-        hasError: props.hasError,
-        placeholder: props.placeholder,
-      }}
-      onChange={onValueChange}
-      plugins={plugins}
-      renderEditor={renderEditor}
-    />
-  );
-};
+  shouldComponentUpdate(nextProps) {
+    let remainingPropsHaveChanged = false;
+    const remainingProps = omit(this.props, propsToOmit);
+
+    // check remaining props normally using !==
+    Object.entries(remainingProps).forEach(([key, val]) => {
+      if (nextProps[key] !== val) remainingPropsHaveChanged = true;
+    });
```

Comment:
This is "just" a shallow equality check, correct?

Reply:
yeah, that would be the default functionality of a "PureComponent", right?

Reply:
Yes, but we still seem to need to do some custom magic on `value.equals`.

Reply:
`value` is the prop that was changing all the time, yes.

Yeah, we need to use immutableJS's `equals` method.

Reply:
Thanks. Got it. Odd though as the whole point of immutable data structures is that there're referential equality holds.

Reply:
Wondering if one would test this.